### PR TITLE
perf: render optimizations

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Improve component performance by using statically defined icons, props, and implementing `useCallback()` where appropropriate.

Note, this PR is non-exhaustive.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A, relates to ongoing performance improvements.

## Risk

Whilst this should be a refactor only, there is a risk something got broken along the way.

## Testing

A general app regression test would be useful.

### Engineering

☝️ 

### Operations

☝️ 

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5504** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5504" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5504?utm_source=stack-comment).

File: src/components/LazyLoadAvatar.tsx [hunk 1]
```diff
@@ -24,7 +27,7 @@ export const LazyLoadAvatar: React.FC<LazyLoadAvatarProps> = ({
         onLoad={() => setImageLoaded(true)}
         src={src}
         size={size}
-        icon={<></>}
+        icon={emptyJsx}
```
Comment (gomesalexandre) [inline src/components/LazyLoadAvatar.tsx:30]:
```suggestion
```

File: src/pages/Dashboard/components/ProfileAvatar/AvatarSelectModal.tsx [hunk 2]
```diff
@@ -177,11 +183,11 @@ export const AvatarSelectModal: React.FC<AvatarSelectModalProps> = props => {
                 {translate('avatar.modal.restoreDefault')}
               </Button>
               <Button
-                width={{ base: 'full', md: 'auto' }}
+                width={buttonWidthProp}
                 height='100%'
                 colorScheme='blue'
                 size='sm-multiline'
-                onClick={() => handleSaveChanges(selected)}
+                onClick={handleSaveChanges}
```
Comment (gomesalexandre) [inline src/pages/Dashboard/components/ProfileAvatar/AvatarSelectModal.tsx:190]: 🧠

File: src/pages/Dashboard/components/ProfileAvatar/AvatarSelectModal.tsx [hunk 3]
```diff
@@ -97,20 +102,21 @@ export const AvatarSelectModal: React.FC<AvatarSelectModalProps> = props => {
     [dispatch, props, walletId],
   )
 
+  const handleSaveChanges = useCallback(() => {
+    if (selected && walletId) {
+      dispatch(nft.actions.setWalletSelectedNftAvatar({ nftAssetId: selected, walletId }))
```
Comment (gomesalexandre) [inline src/pages/Dashboard/components/ProfileAvatar/AvatarSelectModal.tsx:107]: Can leverage `handleChangeAvatar` with `handleChangeAvatar(selectedAvatar)`

**Review by gomesalexandre — COMMENTED** [review]
Conceptual stamp pending a paranoia smoke test, absolute gangster 🐐

**Review by gomesalexandre — APPROVED** [review]
Did a general smoke test and couldn't find anything standing out. Get in ⚡

File: src/pages/Dashboard/components/ProfileAvatar/AvatarSelectModal.tsx (hunk 3 above)
Comment (0xApotheosis) [reply src/pages/Dashboard/components/ProfileAvatar/AvatarSelectModal.tsx:107]: 🧠

**Pull request merged**